slam_toolbox/src/slam_toolbox_common.cpp

```diff
@@ -172,12 +203,20 @@ void SlamToolbox::publishTransformLoop(const double& transform_publish_period)
   {
     {
       boost::mutex::scoped_lock lock(map_to_odom_mutex_);
-      geometry_msgs::TransformStamped msg;
-      tf2::convert(map_to_odom_, msg.transform);
-      msg.child_frame_id = odom_frame_;
-      msg.header.frame_id = map_frame_;
-      msg.header.stamp = ros::Time::now() + transform_timeout_;
-      tfB_->sendTransform(msg);
+      // Update with / add latest transform
+      if(map_to_odom_child_frame_id_.length() > 0)
+        m_map_to_odoms_[map_to_odom_child_frame_id_] = map_to_odom_;
```

Comment:
Left an answer in reply to another one of your comments (in slam_toolbox_common.hpp) that also addresses this. But yes robot1 and robot2 running one after the other could be an issue, though it likely wouldn't matter a whole lot, nor is that issue likely to occur, but I'm looking into a fix to prevent that from being a problem (elaborated on in my other reply)